distributions.py

```diff
@@ -41,82 +41,89 @@ def sample_polar_angle():
             (0) or absorption (1).
- @param     sigma_t the total cross-section of the material region.
- @param     sigma_s the scattering cross-section of the material region.
+ @param     mat in instance of the class Material that contains information
+            about the material            
  @return    An interaction type (0 = scattering, 1 = absorption)
 '''
-def sample_interaction(sigma_t, sigma_s):
-    return int(random.random() < sigma_s / sigma_t):
+def sample_interaction(mat):
+    return int(random.random() < mat.sigma_s / mat.sigma_t)
 
 '''
  @brief     Function that samples a random location within a bounding box.
  @details   A point is randomly and uniformally sampled in the bounding box 
             provided in the input.
- @param     x_min the minimum x-coordinate in the bounding box
- @param     x_max the maximum x-coordinate in the bounding box
- @param     y_min the minimum y-coordinate in the bounding box
- @param     y_max the maximum y-coordinate in the bounding box
- @param     z_min the minimum z-coordinate in the bounding box
- @param     z_max the maximum z-coordinate in the bounding box
- @return    A tuple containing the x, y, and z coordinates of the sampled 
-            location
-'''
-def sample_location(x_min, x_max, y_min, y_max, z_min, z_max):
+ @param     bounds an instance of the boundaries class containing the limits
+            of the bounding box
+ @return    point an instance of the Coords class that contains the coordinates
+            of a point
+'''
+def sample_location(bounds):
+    x_max = bounds.get_x_max()
+    x_min = bounds.get_x_min()
+    y_max = bounds.get_y_max()
+    y_min = bounds.get_y_min()
+    z_max = bounds.get_z_max()
+    z_min = bounds.get_z_min()
+    
     x_pos = x_min + (x_max - x_min) * random.random()
     y_pos = y_min + (y_max - y_min) * random.random()
     z_pos = z_min + (z_max - z_min) * random.random()
-    return (x_pos, y_pos, z_pos)
-
+    point = coords.Coords(x_pos, y_pos, z_pos)
+    return point
 
 '''
  @brief     Function that samples the interaction type given an absorption
             (0 = capture, 1 = fission)
  @details   Based on cross-sections, the interaction is sampled as capture
             (0) or fission (1).
- @param     sigma_a the absorption cross-section of the material region.
- @param     sigma_f the fission cross-section of the material region.
+ @param     mat an instance of the Material class that contains information
+            about the material
  @return    An interaction type (0 = capture, 1 = fission)
 '''
-def sample_fission(simga_a, sigma_f):
-    return int(random.random() < sigma_f / sigma_a):
-
+def sample_fission(mat):
+    return int(random.random() < mat.sigma_f / mat.sigma_a)
 
 '''
  @brief     Samples the nunber of neutrons produced from a fission event
- @param     nu average number of neutrons emitted per fission
+ @param     mat an instance of the Material class that contains information
```

Comment:
ditto about "an instance of ...." ... I'll stop making these comments. Just search for "an instance of ...." and replace with something like "mat a Material object that....".